pytest-embedded-idf/tests/test_idf.py

```diff
@@ -357,6 +358,29 @@ def test_detect_port(dut):
     result.assert_outcomes(passed=1)
 
 
+@pytest.mark.skipif(platform.machine() != 'x86_64', reason='The test is intended to be run on an x86_64 machine.')
+def test_hello_world_linux(testdir):
+    def _change_elf_permissions(path, mode):
```

Comment:
same question, why we're changing the permission here? if the elf file is with executable flag already, could you check who changed this permission?